Instancio errors when trying to create a Set or EnumSet of an enum with entries that have an anonymous class body

**Describe the bug**
Instancio cannot create a class if one of its fields is a Set or EnumSet of an enum whose entries have an anonymous class body.

**Expected result**
For Instancio to create the set with valid entries of the enum.

**Actual result**
If `Keys.FAIL_ON_ERROR` is not enabled, the generated class instance will be `null`. If it is enabled, an exception is thrown:
```
Exception in thread "main" org.instancio.exception.InstancioException: Instancio encountered an error.
Please submit a bug report including the stacktrace:
https://github.com/instancio/instancio/issues
	at org.instancio.internal.util.Fail.withInternalError(Fail.java:50)
	at org.instancio.internal.ErrorHandler.conditionalFailOnError(ErrorHandler.java:71)
	at org.instancio.internal.InstancioEngine.createRootObject(InstancioEngine.java:114)
	at org.instancio.internal.ApiImpl.createRootObject(ApiImpl.java:274)
	at org.instancio.internal.ApiImpl.create(ApiImpl.java:236)
	at org.example.Main.main(Main.java:32)
Caused by: java.lang.ClassCastException: class org.example.MyEnum$1 not an enum
	at java.base/java.util.EnumSet.noneOf(EnumSet.java:115)
Caused by: java.lang.ClassCastException: class org.example.MyEnum$1 not an enum

	at org.instancio.internal.generator.util.EnumSetGenerator.lambda$hints$0(EnumSetGenerator.java:162)
	at org.instancio.internal.InstancioEngine.generateContainer(InstancioEngine.java:720)
	at org.instancio.internal.InstancioEngine.doCreateObject(InstancioEngine.java:203)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:163)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:214)
	at org.instancio.internal.InstancioEngine.generateRecord(InstancioEngine.java:579)
	at org.instancio.internal.InstancioEngine.doCreateObject(InstancioEngine.java:201)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:163)
	at org.instancio.internal.InstancioEngine.createObject(InstancioEngine.java:214)
	at org.instancio.internal.InstancioEngine.createRootObjectInternal(InstancioEngine.java:120)
	at org.instancio.internal.ErrorHandler.conditionalFailOnError(ErrorHandler.java:66)
	... 4 more
```

**Additional information**
Generating a List<MyEnum> works fine.

**Link to a minimal sample project reproducing the issue or runnable sample code**
```java
package org.example;

import org.instancio.Instancio;
import org.instancio.settings.Keys;
import org.instancio.settings.Settings;

import java.util.EnumSet;

record SomeRecord(EnumSet<MyEnum> myEnum) {
}

enum MyEnum {
    // Important: include an anonymous class body, can even be empty
    FIRST {},
}

public class Main {
    public static void main(String[] args) {
        for (int i = 0; i < 1000; i++) {
            Instancio.of(SomeRecord.class)
                    .withSettings(Settings.defaults().set(Keys.FAIL_ON_ERROR, true))
                    .create();
        }
    }
}
```

- Java version: 21
- Instancio version: 5.5.0 (latest as of writing)

--- a/instancio-core/src/main/java/org/instancio/internal/generator/util/EnumSetGenerator.java
+++ b/instancio-core/src/main/java/org/instancio/internal/generator/util/EnumSetGenerator.java
@@ -47,16 +47,17 @@ public class EnumSetGenerator<E extends Enum<E>> extends AbstractGenerator<Set<E
         this.generateEntriesHint = 0;
     }
 
-    @Override
-    public String apiMethod() {
-        return "enumSet()";
-    }
-
+    @SuppressWarnings("unused") // used via reflection
     public EnumSetGenerator(final GeneratorContext context) {
         super(context);
         this.enumClass = null;
         // Without knowing the enum class size cannot be determined, so just default to 1
         this.generateEntriesHint = 1;
+    }
+
+    @Override
+    public String apiMethod() {
+        return "enumSet()";
     }
 
     @Override
@@ -159,10 +160,12 @@ public class EnumSetGenerator<E extends Enum<E>> extends AbstractGenerator<Set<E
         return Hints.builder()
                 .with(InternalContainerHint.builder()
                         .generateEntries(generateEntriesHint)
-                        .createFunction(args -> EnumSet.noneOf((Class<E>) args[0].getClass()))
+                        .createFunction(args -> {
+                            final Class<E> klass = ((E) args[0]).getDeclaringClass();
+                            return EnumSet.noneOf(klass);
+                        })
                         .addFunction((Set<E> enumSet, Object... args) -> enumSet.add((E) args[0]))
                         .build())
                 .build();
     }
-
 }

--- a/instancio-tests/feature-tests/src/test/java/org/instancio/test/features/generator/enumset/GH1413Test.java
+++ b/instancio-tests/feature-tests/src/test/java/org/instancio/test/features/generator/enumset/GH1413Test.java
@@ -1,0 +1,50 @@
+/*
+ * Copyright 2022-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.instancio.test.features.generator.enumset;
+
+import org.instancio.Instancio;
+import org.instancio.TypeToken;
+import org.instancio.junit.InstancioExtension;
+import org.instancio.settings.Keys;
+import org.instancio.test.support.tags.Feature;
+import org.instancio.test.support.tags.FeatureTag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import java.util.EnumSet;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * See https://github.com/instancio/instancio/issues/1413
+ */
+@FeatureTag(Feature.GENERATOR)
+@ExtendWith(InstancioExtension.class)
+class GH1413Test {
+
+    private enum EnumWithAnonymousClass {
+        ANONYMOUS_CLASS_VALUE {}
+    }
+
+    @Test
+    void enumValueAnonymousClass() {
+        final EnumSet<EnumWithAnonymousClass> result = Instancio.of(new TypeToken<EnumSet<EnumWithAnonymousClass>>() {})
+                .withSetting(Keys.FAIL_ON_ERROR, true)
+                .create();
+
+        assertThat(result).containsOnly(EnumWithAnonymousClass.ANONYMOUS_CLASS_VALUE);
+    }
+}

